index.html

```diff
@@ -0,0 +1,261 @@
+              var input = event.target,
+                  value = input.value,
+                  cell = input.parentNode,
+                  rowNum = cell.parentNode.rowIndex - 1;
+              if (value === '') {
+                  value = Number.MAX_SAFE_INTEGER;
+              }
+              tbl[rowNum][1] = parseInt(value);
+          });
+      }
+      function addBooleanFieldEvent(row) {
+          row.cells[2].childNodes[0].addEventListener('change', function (event) {
+              var input = event.target,
+                  checked = input.checked,
+                  cell = input.parentNode,
+                  rowNum = cell.parentNode.rowIndex - 1;
+  
+              tbl[rowNum][2] = checked;
+          });
+      }
+      function addDateFieldEvent(row) {
+          row.cells[3].childNodes[0].addEventListener('change', function (event) {
+              var input = event.target || event.srcElement,
+                  cell = input.parentNode,
+                  rowNum = cell.parentNode.rowIndex - 1,
+                  date = input.value;
+  
+              tbl[rowNum][3] = date;
+          });
+      }
+      function addDeleteFieldEvent(row) {
+          row.cells[4].childNodes[0].addEventListener('click', function (event) {
+              var input = event.target || event.srcElement,
+                  cell = input.parentNode,
+                  rowNum = cell.parentNode.rowIndex - 1,
+                  table = document.querySelector("#tblBody");
+  
+              table.deleteRow(rowNum);
+              tbl.splice(rowNum, 1);
+          });
+      }
+  
+      function assignEvents(row) {
+          addStringFieldEvent(row);
+          addNumberFieldEvent(row);
+          addBooleanFieldEvent(row);
+          addDateFieldEvent(row);
+          addDeleteFieldEvent(row);
+      }
+  
+      function addNameCell(nameCell) {
+          var name = document.createElement('input');
+          name.setAttribute('type', 'text');
+          name.setAttribute('class', 'inputField');
+          nameCell.appendChild(name);
+      }
+  
+      function addNumberCell(numberCell) {
+          var number = document.createElement('input');
+          number.setAttribute('type', 'number');
+          number.setAttribute('class', 'inputField');
+          numberCell.appendChild(number);
+      }
+  
+      function addBooleanCell(booleanCell) {
+          var checkbox = document.createElement('input');
+          checkbox.setAttribute('type', 'checkbox');
+          checkbox.setAttribute('class', 'center');
+          checkbox.checked = false;
+          booleanCell.setAttribute('class', 'btn');
+          booleanCell.appendChild(checkbox);
+      }
+  
+      function addDateCell(dateCell) {
+          var date = document.createElement('input');
+          date.setAttribute('type', 'date');
+          date.setAttribute('name', 'DateCell');
+          date.setAttribute('class', 'center');
```

Comment:
For once, if there are existing classes on your element, they will be overridden, and even if you fetch the classes beforehand and add your class to it (with `curr + " newClass"`), the browser can't optimize because you changed all of the classes, so now it has to recalculate everything from scratch.